library/ssl_tls.c

```diff
@@ -5974,6 +6002,24 @@ static int ssl_session_save( const mbedtls_ssl_session *session,
 #endif /* MBEDTLS_SSL_KEEP_PEER_CERTIFICATE */
 #endif /* MBEDTLS_X509_CRT_PARSE_C */
 
+    if( session == NULL ) return ( MBEDTLS_ERR_SSL_INTERNAL_ERROR );
+
+#if defined(MBEDTLS_SSL_NEW_SESSION_TICKET)
+    /* Ticket format depends on the TLS version negotiated. */
+    minor_ver = session->minor_ver;
+
+#if defined(MBEDTLS_SSL_CLI_C)
+    if( ( session->endpoint == MBEDTLS_SSL_IS_CLIENT ) && ( minor_ver == MBEDTLS_SSL_MINOR_VERSION_4 ) )
+    {
+        /* Check whether we got a ticket already */
```

Comment:
What's the purpose of this check? This function is just serializing data from a C structure, so I think we should it to be called arbitrarily many times on a given `mbedtls_ssl_session` instance. That's different from the new semantics of `mbedtls_ssl_get_session()`, which should now fail if we have already retrieved all tickets (so far: one or none).

Reply:
Probably a wrong thought. Removed it.